exporter/elasticsearchexporter/model.go

```diff
@@ -138,6 +141,15 @@ func (m *encodeModel) encodeLogDefaultMode(resource pcommon.Resource, record plo
 	return document
 }
 
+func (m *encodeModel) encodeLogBodyMapMode(record plog.LogRecord) ([]byte, error) {
+	body := record.Body()
+	if body.Type() != pcommon.ValueTypeMap {
+		return []byte{}, errors.New("record body is not a map")
```

Comment:
@carsonip does this lead to clients re-trying because the batch contains an error? We should ensure this doesn't get re-tried because it's a persistent error.

Reply:
According to my current understanding, yes, although I haven't tried to reproduce the team's report about how upstream components retry es exporter errors

Reply:
What would be the alternative? Serialize to an `{}` empty document?

Reply:
I went ahead and changed to return an empty document if the body is not a map. Let me know if you disagree.

Reply:
sorry for not being clear. For invalid metrics, we silently drop the data point. I believe we want to do the same for log record, but apparently that needs to be done in exporter.go, not model.go, and require a bit of coordination between the two. In that case, what about returning a specific error, and catch it up the call chain to log and drop it?

Reply:
That makes sense. I'll update the code with the changes. Thanks again!